crates/cli-support/src/js/mod.rs

```diff
@@ -1247,7 +1292,7 @@ impl<'a> Context<'a> {
                 fields: Vec::new(),
             })?;
             self.global(&format!("let cached{} = new {}{};", s, name, args));
-        } else if !self.config.mode.always_run_in_browser() {
+        } else if !self.config.mode.always_run_in_browser() && !self.config.mode.deno() {
```

Comment:
the function `always_run_in_browser` seems to only be used to determine whether `const lTextEncoder = typeof TextEncoder === 'undefined' ? (0, module.require)('util').TextEncoder : TextEncoder;` will be used instead of the text processor directly, so maybe renaming the method to `text_processor_available` and including `OutputMode::Deno` may be cleaner.

Reply:
This function is sort of turning into spaghettit with its conditionals, mind just throwing in an explicit `match` over the output mode for this?